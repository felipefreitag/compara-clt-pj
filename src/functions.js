function handleChange(event, employee) {
  const { value } = event.target;
  (employee ? this.setState({ employeeMonthly: value })
    : this.setState({ companyMonthly: value })
  )
}

function handleEmployeeChange(event) {
  this.handleChange(event, true)
}

function handleCompanyChange(event) {
  this.handleChange(event, false)
}

function inssCalc(salary) {
  let result = 0;
  if (salary <= 1659.38 ) {
    result= salary * 0.08
  } else if (salary <= 2765.66) {
    result= salary * 0.09
  } else if (salary <= 5531.31) {
    result= salary * 0.11
  } else result= (5531.31 * 0.11);
  return result;
}

function irrfCalc(salary) {
  let result = 0;
  if (salary <= 1903.98) {
    result = 0
  } else if (salary <= 2826.65) {
    result = (salary * 0.075 - 142.80)
  } else if (salary <= 3751.05) {
    result = (salary * 0.15 - 354.80)
  } else if (salary <= 4664.68) {
    result = (salary * 0.225 - 636.13)
  } else result = (salary * 0.275 - 869.36);
  return result;
}

function transport(salary) {
  const pass = 4.05
  const monthTransport = 4.05 * 2 * 22
  const percent = salary * 0.06
  return monthTransport > percent ? percent : monthTransport
}

function netSalary(salary) {
  const inss = inssCalc(salary);
  const irrf = irrfCalc(salary - inss);
  const netValue = salary - inss - irrf;
  return netValue;
}

function employeeYearlyEarnings(employeeMonthly) {
  const grossEarning = parseFloat(employeeMonthly);
  const netMonthly = netSalary(grossEarning);
  const vacations = netSalary(grossEarning * 1.3);
  // Total: 11 monthly payments + 1 with vacations + 13ยบ
  const yearTotal = 12 * netMonthly + vacations;
  return yearTotal.toLocaleString('de-DE', { maximumFractionDigits: '2'});
}

function employeeResults(employeeMonthly) {
  const grossEarning = parseFloat(employeeMonthly);
  const fgts = grossEarning * 13.3 * 0.08;
  return {
    name: "employee",
    yearly: employeeYearlyEarnings(employeeMonthly),
    fgts: fgts.toLocaleString('de-DE', { maximumFractionDigits: '2'}),
  }
}

function companyYearlyEarnings(companyMonthly) {
  const grossEarning = parseFloat(companyMonthly);
  const netMonthly = grossEarning - 60;
  const yearTotal = 12 * netMonthly;
  return yearTotal.toLocaleString('de-DE', { maximumFractionDigits: '2'});
}

function companyResults(companyMonthly) {
  return {
    name: "company",
    yearly: companyYearlyEarnings(companyMonthly),
  }
}

export { handleChange, handleEmployeeChange, handleCompanyChange }
export { employeeResults, companyResults }
export { inssCalc }
