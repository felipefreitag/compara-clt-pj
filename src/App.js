import React, { Component } from 'react';
import './App.css';
// material-ui
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
// local components
import Page from './Page'
import DataEntry from './DataEntry'
import ShowResults from './ShowResults'
import Disclaimer from './Disclaimer'
// functions
import { handleChange, handleEmployeeChange, handleCompanyChange } from './functions'
import { employeeResults, companyResults } from './functions'

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showResult: false,
      employeeMonthly: 0,
      companyMonthly:  0,
      employeeYearly: 0,
      companyYearly: 0,
      employeeResults: {},
      companyResults: {},
    };

    this.handleChange = handleChange
    this.handleEmployeeChange = handleEmployeeChange
    this.handleCompanyChange = handleCompanyChange
  }

  calculate() {
    this.setState({
      showResult: true,
      employeeResults: employeeResults(this.state.employeeMonthly),
      companyResults: companyResults(this.state.companyMonthly),
    })
  }

  render() {
    return (
      <MuiThemeProvider>
        <div>
          <Page>
            <AppBar
              title="Melhor oferta de trabalho"
              showMenuIconButton={false}
            />
            <h3 style={{width: '100%', textAlign: 'center'}}>
              Compare qual seria seu rendimento anual com CLT x PJ
            </h3>
            <DataEntry
              showResult={this.state.showResult}
              handleEmployeeChange={this.handleEmployeeChange.bind(this)}
              handleCompanyChange={this.handleCompanyChange.bind(this)}
              calculate={this.calculate.bind(this)}
            />
            { this.state.showResult ?
              <div>
                <ShowResults
                  employee={this.state.employeeResults}
                  company={this.state.companyResults}
                />
                <Disclaimer />
              </div>
              :
              <h5>Use apenas números. Clique no botão para ver os resultados.</h5>
            }
          </Page>
        </div>
      </MuiThemeProvider>
    );
  }
}

export default App;
