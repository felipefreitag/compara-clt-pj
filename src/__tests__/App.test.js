import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { employeeResults, companyResults } from '../functions'
import App from '../App'

jest.mock('../functions')

Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div)
})

it('renders correctly', () => {
  const tree = renderer.create(
      <App />
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

describe('calculate', () => {
  const initialState = {
    showResult: false,
    employeeMonthly: 0,
    companyMonthly:  0,
    employeeYearly: 0,
    companyYearly: 0,
    employeeResults: {},
    companyResults: {},
  }
  it('calls the correct functions', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.instance().state).toEqual(initialState)
    wrapper.instance().calculate()
    expect(employeeResults).toHaveBeenCalled()
    expect(companyResults).toHaveBeenCalled()
    expect(wrapper.instance().state.showResult).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
