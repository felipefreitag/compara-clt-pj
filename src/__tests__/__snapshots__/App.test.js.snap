// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "margin": "auto",
        "maxWidth": "960px",
        "width": "96%",
      }
    }
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#00bcd4",
          "borderRadius": "0px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "paddingLeft": 24,
          "paddingRight": 24,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "zIndex": 1100,
        }
      }
    >
      <h1
        onClick={[Function]}
        style={
          Object {
            "boxFlex": 1,
            "color": "#ffffff",
            "flex": "1",
            "fontSize": 24,
            "fontWeight": 400,
            "height": 64,
            "letterSpacing": 0,
            "lineHeight": "64px",
            "margin": 0,
            "muiPrepared": true,
            "overflow": "hidden",
            "paddingTop": 0,
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        Melhor oferta de trabalho
      </h1>
    </div>
    <h3
      style={
        Object {
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Compare qual seria seu rendimento anual com CLT x PJ
    </h3>
    <div
      style={
        Object {
          "height": "auto",
          "muiPrepared": true,
          "overflow": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <table
          className={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderCollapse": "collapse",
              "borderSpacing": 0,
              "fontFamily": "Roboto, sans-serif",
              "muiPrepared": true,
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <thead
            className={undefined}
            style={
              Object {
                "borderBottom": "1px solid #e0e0e0",
                "muiPrepared": true,
              }
            }
          >
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": "1px solid #e0e0e0",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <th
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "color": "#9e9e9e",
                    "fontSize": 12,
                    "fontWeight": "normal",
                    "height": 56,
                    "muiPrepared": true,
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "position": "relative",
                    "textAlign": "center",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                CLT
              </th>
              <th
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "color": "#9e9e9e",
                    "fontSize": 12,
                    "fontWeight": "normal",
                    "height": 56,
                    "muiPrepared": true,
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "position": "relative",
                    "textAlign": "center",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                PJ
              </th>
            </tr>
          </thead>
        </table>
      </div>
      <div
        style={
          Object {
            "height": "inherit",
            "muiPrepared": true,
            "overflowX": "hidden",
            "overflowY": "auto",
          }
        }
      >
        <table
          className={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderCollapse": "collapse",
              "borderSpacing": 0,
              "fontFamily": "Roboto, sans-serif",
              "muiPrepared": true,
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <tbody
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": false,
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  className={undefined}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "cursor": "auto",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 16,
                      "height": 72,
                      "lineHeight": "24px",
                      "muiPrepared": true,
                      "position": "relative",
                      "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className={undefined}
                    htmlFor="Salário bruto CLT"
                    onClick={undefined}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.3)",
                        "lineHeight": "22px",
                        "muiPrepared": true,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 38,
                        "transform": "scale(1) translate(0, 0)",
                        "transformOrigin": "left top",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "zIndex": 1,
                      }
                    }
                  >
                    Salário bruto CLT
                  </label>
                  <input
                    disabled={false}
                    id="Salário bruto CLT"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    style={
                      Object {
                        "WebkitOpacity": 1,
                        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                        "appearance": "textfield",
                        "backgroundColor": "rgba(0,0,0,0)",
                        "border": "none",
                        "boxSizing": "border-box",
                        "color": "rgba(0, 0, 0, 0.87)",
                        "cursor": "inherit",
                        "font": "inherit",
                        "height": "100%",
                        "marginTop": 14,
                        "muiPrepared": true,
                        "outline": "none",
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                    type="text"
                  />
                  <div>
                    <hr
                      aria-hidden="true"
                      style={
                        Object {
                          "borderBottomStyle": "solid",
                          "borderBottomWidth": 1,
                          "borderColor": "#e0e0e0",
                          "borderLeft": "none",
                          "borderRight": "none",
                          "borderTop": "none",
                          "bottom": 8,
                          "boxSizing": "content-box",
                          "margin": 0,
                          "muiPrepared": true,
                          "position": "absolute",
                          "width": "100%",
                        }
                      }
                    />
                    <hr
                      aria-hidden="true"
                      style={
                        Object {
                          "borderBottomStyle": "solid",
                          "borderBottomWidth": 2,
                          "borderColor": "#00bcd4",
                          "borderLeft": "none",
                          "borderRight": "none",
                          "borderTop": "none",
                          "bottom": 8,
                          "boxSizing": "content-box",
                          "margin": 0,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transform": "scaleX(0)",
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                </div>
              </td>
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "left",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  className={undefined}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "cursor": "auto",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "fontSize": 16,
                      "height": 72,
                      "lineHeight": "24px",
                      "muiPrepared": true,
                      "position": "relative",
                      "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className={undefined}
                    htmlFor="Faturamento mensal MEI"
                    onClick={undefined}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.3)",
                        "lineHeight": "22px",
                        "muiPrepared": true,
                        "pointerEvents": "none",
                        "position": "absolute",
                        "top": 38,
                        "transform": "scale(1) translate(0, 0)",
                        "transformOrigin": "left top",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "userSelect": "none",
                        "zIndex": 1,
                      }
                    }
                  >
                    Faturamento mensal MEI
                  </label>
                  <input
                    disabled={false}
                    id="Faturamento mensal MEI"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    style={
                      Object {
                        "WebkitOpacity": 1,
                        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                        "appearance": "textfield",
                        "backgroundColor": "rgba(0,0,0,0)",
                        "border": "none",
                        "boxSizing": "border-box",
                        "color": "rgba(0, 0, 0, 0.87)",
                        "cursor": "inherit",
                        "font": "inherit",
                        "height": "100%",
                        "marginTop": 14,
                        "muiPrepared": true,
                        "outline": "none",
                        "padding": 0,
                        "position": "relative",
                        "width": "100%",
                      }
                    }
                    type="text"
                  />
                  <div>
                    <hr
                      aria-hidden="true"
                      style={
                        Object {
                          "borderBottomStyle": "solid",
                          "borderBottomWidth": 1,
                          "borderColor": "#e0e0e0",
                          "borderLeft": "none",
                          "borderRight": "none",
                          "borderTop": "none",
                          "bottom": 8,
                          "boxSizing": "content-box",
                          "margin": 0,
                          "muiPrepared": true,
                          "position": "absolute",
                          "width": "100%",
                        }
                      }
                    />
                    <hr
                      aria-hidden="true"
                      style={
                        Object {
                          "borderBottomStyle": "solid",
                          "borderBottomWidth": 2,
                          "borderColor": "#00bcd4",
                          "borderLeft": "none",
                          "borderRight": "none",
                          "borderTop": "none",
                          "bottom": 8,
                          "boxSizing": "content-box",
                          "margin": 0,
                          "muiPrepared": true,
                          "position": "absolute",
                          "transform": "scaleX(0)",
                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          "width": "100%",
                        }
                      }
                    />
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <table
          className={undefined}
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderCollapse": "collapse",
              "borderSpacing": 0,
              "fontFamily": "Roboto, sans-serif",
              "muiPrepared": true,
              "tableLayout": "fixed",
              "width": "100%",
            }
          }
        >
          <tfoot
            className={undefined}
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            <tr
              className={undefined}
              style={
                Object {
                  "borderBottom": false,
                  "borderTop": "0px",
                  "color": "rgba(0, 0, 0, 0.87)",
                  "height": 48,
                  "muiPrepared": true,
                  "padding": 20,
                  "textAlign": "left",
                  "verticalAlign": "bottom",
                  "whiteSpace": "nowrap",
                }
              }
            >
              <td
                className={undefined}
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "inherit",
                    "fontSize": 13,
                    "height": 48,
                    "muiPrepared": true,
                    "overflow": "hidden",
                    "paddingLeft": 24,
                    "paddingRight": 24,
                    "textAlign": "center",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                <div
                  className={undefined}
                  style={
                    Object {
                      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                      "backgroundColor": "#ffffff",
                      "borderRadius": 2,
                      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                             0 1px 4px rgba(0, 0, 0, 0.12)",
                      "boxSizing": "border-box",
                      "color": "rgba(0, 0, 0, 0.87)",
                      "display": "inline-block",
                      "fontFamily": "Roboto, sans-serif",
                      "minWidth": 88,
                      "muiPrepared": true,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    }
                  }
                >
                  <button
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                      Object {
                        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                        "backgroundColor": "#00bcd4",
                        "border": 10,
                        "borderRadius": 2,
                        "boxSizing": "border-box",
                        "cursor": "pointer",
                        "display": "inline-block",
                        "fontFamily": "Roboto, sans-serif",
                        "fontSize": "inherit",
                        "fontWeight": "inherit",
                        "height": 36,
                        "lineHeight": "36px",
                        "margin": 0,
                        "muiPrepared": true,
                        "outline": "none",
                        "padding": 0,
                        "position": "relative",
                        "textAlign": "center",
                        "textDecoration": "none",
                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                        "verticalAlign": null,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="button"
                  >
                    <div
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <div
                        style={
                          Object {
                            "backgroundColor": false,
                            "borderRadius": 2,
                            "height": 36,
                            "muiPrepared": true,
                            "top": 0,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "color": "#ffffff",
                              "fontSize": 14,
                              "fontWeight": 500,
                              "letterSpacing": 0,
                              "margin": 0,
                              "muiPrepared": true,
                              "opacity": 1,
                              "paddingLeft": 16,
                              "paddingRight": 16,
                              "position": "relative",
                              "textTransform": "uppercase",
                              "userSelect": "none",
                            }
                          }
                        >
                          Calcular
                        </span>
                      </div>
                    </div>
                  </button>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
    <h5>
      Use apenas números. Clique no botão para ver os resultados.
    </h5>
  </div>
</div>
`;
